### ISSUE TITLE
Roller shutter percentage?

### ISSUE THREAD

**Issue by lassem** [ruebox/openhab2-addons#21]
Both the web-app and the Android app has the possibility of setting the roller shutters to a percentage value. 70% closed f.ex.

Feasible?

**Comment by ruebox** [comment]
Depends. If it can be simply set by a command, this is not too difficult. Would be great if you can sniff the commands via firefox - WebDeveloper as described in #9.

What you could do is the following:
    Got to your freeathome SysAP webpage with firefox
    Open the menu "Web Developer" -> Network (Hot key: Ctrl-Shift-E)
    Reload
    Login to SysAP
    Go to the Freeathome webpage and to the control of your blinds
    Perform the operation
    WebDeveloperWindow: Right-click one log entry -> Save All as HAR
    Upload HAR and a description what you did to github.

I would then check, which commands are transfer to SysAP.

Thx.

**Comment by lassem** [comment]
While I couldn't get this to work in Firefox, Chrome did the job:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<iq xmlns="jabber:client" to="mrha@busch-jaeger.de/rpc" type="set" id="1522007914448">
   <query xmlns="jabber:iq:rpc">
      <methodCall>
         <methodName>RemoteInterface.setDatapoint</methodName>
         <params>
            <param>
               <value>
                  <string>ABB25B712051/ch0000/idp0002</string>
               </value>
            </param>
            <param>
               <value>
                  <string>48</string>
               </value>
            </param>
         </params>
      </methodCall>
   </query>
</iq>
```

**Comment by lassem** [comment]
In the android app and web-app, you can access these additional functions by long-pressing on the icon-buttons.

![image](https://user-images.githubusercontent.com/116084/37879359-06c8f33c-3078-11e8-8900-e57f716ead86.png)

**Comment by ruebox** [comment]
Cool a new feature that I was not aware of.
Will be done for next update. 
Nevertheless, the feature freeze for the first release seems to be on the horizon :-)

**Comment by TheSpirit** [comment]
i tried it with my rollershutter with deviceID 1013:
it works the same way:
`"text": "<body rid='783246' sid='fc9ac387-65b9-411a-89c7-30d01f54864c' xmlns='http://jabber.org/protocol/httpbind' key='50ca492a870e0369a96c2464ebf09471041e1566'><iq xmlns=\"jabber:client\" to=\"mrha@busch-jaeger.de/rpc\" type=\"set\" id=\"1522076768785\"><query xmlns=\"jabber:iq:rpc\"><methodCall><methodName>RemoteInterface.setDatapoint</methodName><params><param><value><string>ABB700C9CE93/ch0003/idp0002</string></value></param><param><value><string>30</string></value></param></params></methodCall></query></iq></body>"
`
only the channel is again ch0003

**Comment by ruebox** [comment]
implemented and tested -> works :+1:  will come with the next update

**Comment by TheSpirit** [comment]
Perfect. When will we get the next update to test?

**Comment by N2641D** [comment]
In order to understand some of the Free@Home communication details and before looking into openHAB, I have started to play with the GitHub software from Stefan Stadlberger. No problems to switch light on and off, run roller shutters up and down.
I could not yet figure out how to move the roller shutter to a certain percentage. I have recognized that I have to read "... ch0003/odp0001" for its current position and to set "... ch0003/idp0002/77" as 77%, however, I'm stuck with my final step to move the shutter. As soon as I send any command (up, down, toggle_up, ...) the 77% will not be recognized.
What am I doing wrong? Can you please help?

Thanks a lot!

**Comment by lassem** [comment]
Are you sure that channel is correct? What device id do you have?
Operating on ch0003 implies that have a blind actuator with at least 4 channels. Is that correct?

An easy test would be to just try these things on ch0000, ch0001 and ch0002 as well.

**Comment by N2641D** [comment]
From what I understand is it a 1 channel device (Jalousieaktor 1/1-fach), however, it just has ch0003.

Here the full dataset of one of my devices:
{"serialNumber":"ABB700C9AB60","deviceId":"1013","typeName":"Sensor/ Jalousieaktor 1/1-fach","channels":{"ch0003":{"datapoints":{"idp0000":"0","idp0001":"1","idp0002":"0","idp0003":"0","idp0008":"0","odp0000":"1","odp0001":"100","odp0002":"0","odp0003":"0","odp0004":"0","pm0000":"14","pm0001":"13","pm0002":"0","pm0004":"2","pm0005":"2","pm0006":"2","pm0007":"1","pm0008":"0","pm0009":"1","pm000A":"1","pm000C":"2","pm000D":"1","pm0011":"3","pm0012":"2"}}}}

I have not yet understood how it should work? I mentioned that I'm using Stefan Stadlberger's software. e.g. "up" is initiated by: http://myip:myport/set/shutter/ABB700C9AB60/ch0003/up. I have no idea what I would need to send to move the device to a pre-defined position.

Thank you!

**Comment by lassem** [comment]
In his library in the file `sysap-external.js` on line 67-73 somehwhere (the shutter actions block), you could try to add:
`'set' : { 'idp0002' : value },`

This would effectively add a new action to the shutters so that you could do:
`http://myip:myport/set/shutter/ABB700C9AB60/ch0003/set/77`

**Comment by N2641D** [comment]
Hi Lasse - It works!

I did the change and it did work strait away. Following your guidance, I now understand a little bit more, how my http command is translated by Stefan's software.

I have 4 FIBARO HC2 systems and 1 Free@Home system at various (private) places. While I have used almost 100% FIBARO in the past 2 years, I understand that you and other home automation experts are starting to use openHAB as a platform to connect with many systems. Although I'm much more a C++ programmer, that change requires me to better look into Java Script as well :-)

Thank you very much for your help!

Wilfried
